qiskit/converters/circuit_to_instruction.py

```diff
@@ -95,15 +97,33 @@ def find_bit_position(bit):
 
     definition = target.data
 
+    regs = []
     if instruction.num_qubits > 0:
         q = QuantumRegister(instruction.num_qubits, 'q')
+        regs.append(q)
+
     if instruction.num_clbits > 0:
         c = ClassicalRegister(instruction.num_clbits, 'c')
+        regs.append(c)
 
     definition = list(map(lambda x:
                           (x[0],
                            list(map(lambda y: q[find_bit_position(y)], x[1])),
                            list(map(lambda y: c[find_bit_position(y)], x[2]))), definition))
-    instruction.definition = definition
+
+    # fix condition
+    for rule in definition:
+        condition = rule[0].condition
+        if condition:
+            reg, val = condition
+            if reg.size == c.size:
+                rule[0].condition = (c, val)
+            else:
+                raise QiskitError('Cannot convert condition in circuit with '
+                                  'multiple classical registers to instruction')
```

Comment:
is this addressing a bug? if so can you add a test?

Reply:
Without this circuit.test_circuit_operations.TestCircuitOperations.test_repeat and circuit.test_instructions.TestInstructions.test_reverse_instruction fail.